pkg/vex/vex.go

```diff
@@ -34,29 +34,28 @@ func FromPackageConfiguration(buildCfg build.Configuration, vexCfg Config) (vex.
 	return doc, nil
 }
 
-func statementsFromConfiguration(cfg build.Configuration, documentTimestamp time.Time, purls []string) []vex.Statement {
+func statementsFromConfiguration(cfg *build.Configuration, documentTimestamp time.Time, purls []string) []vex.Statement {
 	// We should also add a lint rule for when advisories obviate particular secfixes items.
 	secfixesStatements := statementsFromSecfixes(cfg.Secfixes, purls)
 	advisoriesStatements := statementsFromAdvisories(cfg.Advisories, purls)
 
 	// don't include "not_affected" statements from secfixes that are obviated by statements from advisories
 	notAffectedVulns := make(map[string]struct{})
-	for _, stmt := range advisoriesStatements {
-		if stmt.Status == vex.StatusNotAffected {
-			notAffectedVulns[stmt.Vulnerability] = struct{}{}
+	for i := range advisoriesStatements {
+		if advisoriesStatements[i].Status == vex.StatusNotAffected {
```

Comment:
Why change to using index integers instead of using the slice elements directly?